### ISSUE TITLE
Can't update flake

### ISSUE THREAD

**Issue by ptonner** [snowfallorg/lib#28]
After running a `nix flake update`, I get an error trying to build my system:

```sh
 Failed assertions:
       - Your system configures nixpkgs with an externally created instance.
       `nixpkgs.config` options should be passed when creating the instance instead.
```

Full error:
```
error:
       … while calling the 'head' builtin

         at /nix/store/4jfc6vrkmq7z5pb651jh5b4kra5f1kwp-source/lib/attrsets.nix:850:11:

          849|         || pred here (elemAt values 1) (head values) then
          850|           head values
             |           ^
          851|         else

       … while evaluating the attribute 'value'

         at /nix/store/4jfc6vrkmq7z5pb651jh5b4kra5f1kwp-source/lib/modules.nix:807:9:

          806|     in warnDeprecation opt //
          807|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          808|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error:
       Failed assertions:
       - Your system configures nixpkgs with an externally created instance.
       `nixpkgs.config` options should be passed when creating the instance instead.

       Current value:
       {
         __toString = <function>;
         _type = "option";
         declarationPositions = [
           {
             column = 5;
             file = "/nix/store/4jfc6vrkmq7z5pb651jh5b4kra5f1kwp-source/nixos/modules/misc/nixpkgs.nix";
             line = 148;
           }
         ];
         declarations = [
           "/nix/store/4jfc6vrkmq7z5pb651jh5b4kra5f1kwp-source/nixos/modules/misc/nixpkgs.nix"
         ];
         default = { };
         definitions = [
           {
             allowUnfree = true;
           }
         ];
         definitionsWithLocations = [
           {
             file = "/nix/store/rc9d9bp3fal596xin3mksgkf8car71bj-m4hfm2ry6b3c00lndcvp5x6g60sfk35q-source/systems/x86_64-linux/msi-b450/default.nix";
             value = {
               allowUnfree = true;
             };
           }
         ];
         description = ''
           The configuration of the Nix Packages collection.  (For
           details, see the Nixpkgs documentation.)  It allows you to set
           package configuration options.

           Ignored when `nixpkgs.pkgs` is set.
         '';
         example = {
           _type = "literalExpression";
           text = ''
             { allowBroken = true; allowUnfree = true; }
           '';
         };
         files = [
           "/nix/store/rc9d9bp3fal596xin3mksgkf8car71bj-m4hfm2ry6b3c00lndcvp5x6g60sfk35q-source/systems/x86_64-linux/msi-b450/default.nix"
         ];
         highestPrio = 100;
         isDefined = true;
         loc = [
           "nixpkgs"
           "config"
         ];
         options = [ ];
         type = {
           _type = "option-type";
           check = <function>;
           deprecationMessage = null;
           description = "nixpkgs config";
           descriptionClass = null;
           emptyValue = { };
           functor = {
             binOp = <function>;
             name = "nixpkgs-config";
             payload = null;
             type = null;
             wrapped = null;
           };
           getSubModules = null;
           getSubOptions = <function>;
           merge = <function>;
           name = "nixpkgs-config";
           nestedTypes = { };
           substSubModules = <function>;
           typeMerge = <function>;
         };
         value = {
           allowUnfree = true;
         };
       }
```

**Comment by ptonner** [comment]
does snowfall work on unstable? if not that might be the issue

**Comment by jakehamilton** [comment]
@ptonner have you updated both Snowfall Lib and NixPkgs to the latest version? There was previously an issue with flake utils due to a change in NixPkgs.

**Comment by jakehamilton** [comment]
This should have been fixed: https://github.com/snowfallorg/lib/issues/24

**Comment by ptonner** [comment]
Is there anything in particular I need to do beyond a `nix flake update`? I can try to setup a minimal example at some point, b/c my config isn't public at the momement. But the basic setup is

```nix
{
   inputs = {
      nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
      snowfall-lib = {
        url = "github:snowfallorg/lib";
        inputs.nixpkgs.follows = "nixpkgs";
      };
      home-manager = {
        url = "github:nix-community/home-manager";
        inputs.nixpkgs.follows = "nixpkgs";
      };
      # ...
   };
}
```

and trying a `sudo nixos-rebuild switch --flake .` raises the error, even after a `nix flake update`

**Comment by ptonner** [comment]
I think I may have finally tracked this down to a `nixpkgs.config.allowUnfree = true;` in one of my nixos modules. I removed that line and changing my `mkFlake` call to:

```nix
    lib.mkFlake {
      channels-config = {
        allowUnfree = true;
      };
    };
```

Everything seems to work after that.



The channels-config part I got from [khaneliman's config](https://github.com/khaneliman/khanelinix/blob/e8e1894533adb880fbdd2dc03b2a85b4e33187fb/flake.nix#L178), and AFAICT there's nothing about `channels-config` in the docs, might be something to add?

Anyways glad to have things working again and thanks for the great tool!
